File: appliboard/src/main/java/com/backend/appliboard/features/analytics/dto/AnalyticsDto.java
package com.backend.appliboard.features.analytics.dto;

import com.backend.appliboard.features.job_application.Status;

import java.util.Map;

public record AnalyticsDto(
        Long totalApplicationsSent,
        Long weeklyApplicationsSent,
        Map<Status, Long> countByStatus
) {
}

File: frontend/src/features/dashboard/analytics/models/AnalyticsDto.ts
import type { Status } from "../../applications/models/Status";

export type DayOfWeek =
    | "SUNDAY"
    | "MONDAY"
    | "TUESDAY"
    | "WEDNESDAY"
    | "THURSDAY"
    | "FRIDAY"
    | "SATURDAY";

export interface AnalyticsDto {

    totalApplicationsSent: number;
    weeklyApplicationsSent: number;
    countByStatus: Record<Status, number>;
    weeklyApplicationsByDay: Record<DayOfWeek, number>;
    
}